File: admin_app/admin_flutter/lib/models/master_finance_setting.dart
class MasterFinanceSetting {
  const MasterFinanceSetting({
    required this.enabled,
    required this.feePercent,
    required this.dailyInterestPercent,
    required this.graceDays,
    required this.monetaryCorrectionEnabled,
    required this.monetaryIndex,
    required this.automaticIndexUpdate,
  });
  final bool enabled;
  final double feePercent;
  final double dailyInterestPercent;
  final int graceDays;
  final bool monetaryCorrectionEnabled;
  final String monetaryIndex;
  final bool automaticIndexUpdate;
  factory MasterFinanceSetting.fromJson(Map<String, dynamic> j) =>
      MasterFinanceSetting(
        enabled: j['late_charges_enabled'] == true,
        feePercent: (j['late_fee_percent'] as num?)?.toDouble() ?? 0,
        dailyInterestPercent:
            (j['late_interest_daily_percent'] as num?)?.toDouble() ?? 0,
        graceDays: (j['late_grace_days'] as num?)?.toInt() ?? 0,
        monetaryCorrectionEnabled: j['monetary_correction_enabled'] != false,
        monetaryIndex: j['monetary_index'] as String? ?? 'IPCA',
        automaticIndexUpdate: j['automatic_index_update'] != false,
      );
  Map<String, dynamic> toJson() => {
    'late_charges_enabled': enabled,
    'late_fee_percent': feePercent,
    'late_interest_daily_percent': dailyInterestPercent,
    'late_grace_days': graceDays,
    'monetary_correction_enabled': monetaryCorrectionEnabled,
    'monetary_index': monetaryIndex,
    'automatic_index_update': automaticIndexUpdate,
  };
}

File: backend/app/schemas/master_finance_setting.py
from datetime import datetime
from pydantic import BaseModel, Field


class MasterFinanceSettingRead(BaseModel):
    late_charges_enabled: bool
    late_fee_percent: float
    late_interest_daily_percent: float
    late_grace_days: int
    monetary_correction_enabled: bool
    monetary_index: str
    automatic_index_update: bool
    updated_at: datetime | None = None


class MasterFinanceSettingUpdate(BaseModel):
    late_charges_enabled: bool = False
    late_fee_percent: float = Field(0, ge=0, le=100)
    late_interest_daily_percent: float = Field(0, ge=0, le=10)
    late_grace_days: int = Field(0, ge=0, le=365)
    monetary_correction_enabled: bool = True
    monetary_index: str = Field("IPCA", pattern="^[A-Z0-9_-]{2,20}$")
    automatic_index_update: bool = True
